SpiderFoot.fetchUrl() - Unexpected exception (No connection adapters were found for 'irc://irc.lfnet.org:6667')

`fetchUrl()` should not attempt to retrieve links which do not have a `http[s]` URL scheme.

```
2020-09-09 04:45:54,212 [ERROR] Unexpected exception (No connection adapters were found for 'irc://irc.lfnet.org:6667') occurred fetching URL: irc://irc.lfnet.org:6667
2020-09-09 04:45:54,213 [ERROR] Traceback (most recent call last):
  File "/root/Desktop/spiderfoot/sflib.py", line 2542, in fetchUrl
    hdr = self.getSession().head(
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 568, in head
    return self.request('HEAD', url, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 640, in send
    adapter = self.get_adapter(url=request.url)
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 731, in get_adapter
    raise InvalidSchema("No connection adapters were found for '%s'" % url)
requests.exceptions.InvalidSchema: No connection adapters were found for 'irc://irc.lfnet.org:6667'
```

--- a/sflib.py
+++ b/sflib.py
@@ -2580,6 +2580,16 @@ class SpiderFoot:
 
         url = url.strip()
 
+        try:
+            parsed_url = urllib.parse.urlparse(url)
+        except Exception:
+            self.debug(f"Could not parse URL: {url}")
+            return None
+
+        if parsed_url.scheme != 'http' and parsed_url.scheme != 'https':
+            self.debug(f"Invalid URL scheme for URL: {url}")
+            return None
+
         proxies = dict()
         if self.useProxyForUrl(url):
             proxy_url = f"socks5h://{self.opts['_socks2addr']}:{self.opts['_socks3port']}"

--- a/test/unit/test_spiderfoot.py
+++ b/test/unit/test_spiderfoot.py
@@ -1467,11 +1467,32 @@ class TestSpiderFoot(unittest.TestCase):
         """
         sf = SpiderFoot(self.default_options)
 
-        invalid_types = [None, "", list(), dict(), int()]
+        invalid_types = [None, list(), dict(), int()]
         for invalid_type in invalid_types:
             with self.subTest(invalid_type=invalid_type):
                 res = sf.fetchUrl(invalid_type)
                 self.assertEqual(None, res)
+
+    def test_fetchUrl_argument_url_invalid_url_should_return_None(self):
+        """
+        Test fetchUrl(self, url, fatal=False, cookies=None, timeout=30,
+                 useragent="SpiderFoot", headers=None, noLog=False,
+                 postData=None, dontMangle=False, sizeLimit=None,
+                 headOnly=False, verify=False)
+        """
+        sf = SpiderFoot(self.default_options)
+
+        res = sf.fetchUrl("")
+        self.assertEqual(None, res)
+
+        res = sf.fetchUrl("://spiderfoot.net/")
+        self.assertEqual(None, res)
+
+        res = sf.fetchUrl("file:///etc/hosts")
+        self.assertEqual(None, res)
+
+        res = sf.fetchUrl("irc://spiderfoot.net:6697/")
+        self.assertEqual(None, res)
 
     def test_check_dns_wildcard_invalid_target_should_return_none(self):
         """

